src/error/syntaxError.js

```diff
@@ -33,6 +34,25 @@ export function syntaxError(
 }
 
 /**
+ * Produces a string for the invarant(...) function that renders the location
+ * where an error occurred. If no source is passed in, it renders the message
+ * without context.
+ */
+export function invariantError(
+  message: string,
+  node: ASTNode,
+  source: ?Source
+): string {
+  const position = node.loc ? node.loc.start : null;
+  if (position == null || source == null) {
+    return message;
+  }
+  const location = getLocation(source, position);
+  return `${message} (${location.line}:${location.column}) ` +
```

Comment:
@asiandrummer This prints strange output from the `invariant` function, which inlined would be:

```
if (!condition) {
    throw new Error(new GraphQLError(...));
}
```

But, I could change `invariant(...)` to check first if its message argument is a string, and if it's actually an error, just rethrow. Then I could make `validationError` return a `GraphQLError`. Sensible?